torchgen/native_function_generation.py

```diff
@@ -103,27 +131,42 @@ def self_to_out_signature(func: FunctionSchema) -> FunctionSchema:
     )
 
 
-# Helper function: given a mutable FunctionSchema, generate its corresponding out= variant
+# Helper function: given a functional FunctionSchema, generate its corresponding out= variant
 # Example before:
-#   _fused_moving_avg_obs_fq_helper(Tensor self, Tensor observer_on, Tensor fake_quant_on, Tensor(a!) running_min, Tensor(b!) running_max, Tensor(c!) scale, Tensor(d!) zero_point, float averaging_const, int quant_min, int quant_max, int ch_axis, bool per_row_fake_quant=False, bool symmetric_quant=False) -> (Tensor output, Tensor mask)  # noqa: B950
+#   _to_copy(Tensor self, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None,
+#       bool? pin_memory=None, bool non_blocking=False, MemoryFormat? memory_format=None) -> Tensor
 # Example after:
-#   _fused_moving_avg_obs_fq_helper.out(Tensor self, Tensor observer_on, Tensor fake_quant_on, Tensor(a!) running_min, Tensor(b!) running_max, Tensor(c!) scale, Tensor(d!) zero_point, float averaging_const, int quant_min, int quant_max, int ch_axis, bool per_row_fake_quant=False, bool symmetric_quant=False, *, Tensor(e!) out0, Tensor(f!) out1) -> (Tensor(e!), Tensor(f!))  # noqa: B950
-def mutable_to_out_signature(func: FunctionSchema) -> FunctionSchema:
-    # Generating an out= schema from a mutable schema.
-    assert func.kind() == SchemaKind.mutable
+#   _to_copy.out(Tensor self, *, bool non_blocking=False, MemoryFormat? memory_format=None,
+#       Tensor(a!) out) -> Tensor(a!)
+def functional_to_out_signature(func: FunctionSchema) -> FunctionSchema:
+    # Generating an out= schema from a functional schema.
+    assert func.kind() == SchemaKind.functional
+
+    new_returns, new_out_args = generate_out_args_from_schema(func)
     # The new out= schema has:
-    # - Any non-aliased tensor-like returns are converted to mutable, aliased out= arguments
-    #   (if the argument is a tensor then we also return it for method chaining,
-    #   otherwise we return nothing)
+    # - a new out argument with the same type as "func" (but with a mutable annotation)
```

Comment:
What's "func" here? (maybe this is referring to the return argument in the functional op)